From b4e25ebc165756de912e37f8f03463892df98576 Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Sat, 11 Apr 2020 14:45:02 +0300
Subject: [PATCH 121/133] add custom dark mode overlays

	modified:   packages/SystemUI/src/com/android/systemui/theme/ThemeOverlayManager.java
	modified:   packages/overlays/Android.mk
	new file:   packages/overlays/CharcoalDarkUIOverlay/Android.mk
	new file:   packages/overlays/CharcoalDarkUIOverlay/AndroidManifest.xml
	new file:   packages/overlays/CharcoalDarkUIOverlay/res/values-night/colors.xml
	new file:   packages/overlays/MidnightDarkUIOverlay/Android.mk
	new file:   packages/overlays/MidnightDarkUIOverlay/AndroidManifest.xml
	new file:   packages/overlays/MidnightDarkUIOverlay/res/values-night/colors.xml
---
 .../systemui/theme/ThemeOverlayManager.java   |  8 +++-
 packages/overlays/Android.mk                  |  2 +
 .../overlays/CharcoalDarkUIOverlay/Android.mk | 30 +++++++++++++
 .../CharcoalDarkUIOverlay/AndroidManifest.xml | 24 +++++++++++
 .../res/values-night/colors.xml               | 42 +++++++++++++++++++
 .../overlays/MidnightDarkUIOverlay/Android.mk | 30 +++++++++++++
 .../MidnightDarkUIOverlay/AndroidManifest.xml | 24 +++++++++++
 .../res/values-night/colors.xml               | 42 +++++++++++++++++++
 8 files changed, 201 insertions(+), 1 deletion(-)
 create mode 100644 packages/overlays/CharcoalDarkUIOverlay/Android.mk
 create mode 100644 packages/overlays/CharcoalDarkUIOverlay/AndroidManifest.xml
 create mode 100644 packages/overlays/CharcoalDarkUIOverlay/res/values-night/colors.xml
 create mode 100644 packages/overlays/MidnightDarkUIOverlay/Android.mk
 create mode 100644 packages/overlays/MidnightDarkUIOverlay/AndroidManifest.xml
 create mode 100644 packages/overlays/MidnightDarkUIOverlay/res/values-night/colors.xml

diff --git a/packages/SystemUI/src/com/android/systemui/theme/ThemeOverlayManager.java b/packages/SystemUI/src/com/android/systemui/theme/ThemeOverlayManager.java
index 693903e5aeb..204bf2da88b 100644
--- a/packages/SystemUI/src/com/android/systemui/theme/ThemeOverlayManager.java
+++ b/packages/SystemUI/src/com/android/systemui/theme/ThemeOverlayManager.java
@@ -73,6 +73,9 @@ class ThemeOverlayManager {
     @VisibleForTesting
     static final String OVERLAY_CATEGORY_QSPANEL =
             "android.theme.customization.qspanel";
+    @VisibleForTesting
+    static final String OVERLAY_CATEGORY_DARKUI =
+            "android.theme.customization.darkui";
 
     /*
      * All theme customization categories used by the system, in order that they should be applied,
@@ -85,6 +88,7 @@ class ThemeOverlayManager {
             OVERLAY_CATEGORY_COLOR,
             OVERLAY_CATEGORY_STATUSBAR,
             OVERLAY_CATEGORY_QSPANEL,
+            OVERLAY_CATEGORY_DARKUI,
             OVERLAY_CATEGORY_ICON_ANDROID,
             OVERLAY_CATEGORY_ICON_SYSUI,
             OVERLAY_CATEGORY_ICON_SETTINGS,
@@ -98,6 +102,7 @@ class ThemeOverlayManager {
             OVERLAY_CATEGORY_SHAPE,
             OVERLAY_CATEGORY_QSPANEL,
             OVERLAY_CATEGORY_STATUSBAR,
+            OVERLAY_CATEGORY_DARKUI,
             OVERLAY_CATEGORY_ICON_ANDROID,
             OVERLAY_CATEGORY_ICON_SYSUI);
 
@@ -118,7 +123,7 @@ class ThemeOverlayManager {
         mThemePickerPackage = themePickerPackage;
         mTargetPackageToCategories.put(ANDROID_PACKAGE, Sets.newHashSet(
                 OVERLAY_CATEGORY_COLOR, OVERLAY_CATEGORY_FONT, OVERLAY_CATEGORY_STATUSBAR,
-                OVERLAY_CATEGORY_SHAPE, OVERLAY_CATEGORY_ICON_ANDROID));
+                OVERLAY_CATEGORY_SHAPE, OVERLAY_CATEGORY_ICON_ANDROID, OVERLAY_CATEGORY_DARKUI));
         mTargetPackageToCategories.put(SYSUI_PACKAGE,
                 Sets.newHashSet(OVERLAY_CATEGORY_ICON_SYSUI, OVERLAY_CATEGORY_QSPANEL));
         mTargetPackageToCategories.put(SETTINGS_PACKAGE,
@@ -132,6 +137,7 @@ class ThemeOverlayManager {
         mCategoryToTargetPackage.put(OVERLAY_CATEGORY_SHAPE, ANDROID_PACKAGE);
         mCategoryToTargetPackage.put(OVERLAY_CATEGORY_STATUSBAR, ANDROID_PACKAGE);
         mCategoryToTargetPackage.put(OVERLAY_CATEGORY_ICON_ANDROID, ANDROID_PACKAGE);
+        mCategoryToTargetPackage.put(OVERLAY_CATEGORY_DARKUI, ANDROID_PACKAGE);
         mCategoryToTargetPackage.put(OVERLAY_CATEGORY_QSPANEL, SYSUI_PACKAGE);
         mCategoryToTargetPackage.put(OVERLAY_CATEGORY_ICON_SYSUI, SYSUI_PACKAGE);
         mCategoryToTargetPackage.put(OVERLAY_CATEGORY_ICON_SETTINGS, SETTINGS_PACKAGE);
diff --git a/packages/overlays/Android.mk b/packages/overlays/Android.mk
index 27cef06776d..3c9367d104c 100644
--- a/packages/overlays/Android.mk
+++ b/packages/overlays/Android.mk
@@ -81,6 +81,8 @@ LOCAL_REQUIRED_MODULES := \
 	QSPanel34Overlay \
 	QSPanel43Overlay \
 	QSPanel55Overlay \
+	CharcoalBlackOverlay \
+	MidnightDarkUIOverlay \
 	NavigationBarMode3ButtonOverlay \
 	NavigationBarMode2ButtonOverlay \
 	NavigationBarModeGesturalOverlay \
diff --git a/packages/overlays/CharcoalDarkUIOverlay/Android.mk b/packages/overlays/CharcoalDarkUIOverlay/Android.mk
new file mode 100644
index 00000000000..a2979e5cc84
--- /dev/null
+++ b/packages/overlays/CharcoalDarkUIOverlay/Android.mk
@@ -0,0 +1,30 @@
+#
+#  Copyright 2019, The Android Open Source Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+LOCAL_PATH:= $(call my-dir)
+include $(CLEAR_VARS)
+
+LOCAL_RRO_THEME := CharcoalDarkUI
+
+LOCAL_PRODUCT_MODULE := true
+
+LOCAL_SRC_FILES := $(call all-subdir-java-files)
+
+LOCAL_RESOURCE_DIR := $(LOCAL_PATH)/res
+
+LOCAL_PACKAGE_NAME := CharcoalDarkUIOverlay
+LOCAL_SDK_VERSION := current
+
+include $(BUILD_RRO_PACKAGE)
diff --git a/packages/overlays/CharcoalDarkUIOverlay/AndroidManifest.xml b/packages/overlays/CharcoalDarkUIOverlay/AndroidManifest.xml
new file mode 100644
index 00000000000..d03ba26d5b2
--- /dev/null
+++ b/packages/overlays/CharcoalDarkUIOverlay/AndroidManifest.xml
@@ -0,0 +1,24 @@
+<!--
+/**
+ * Copyright (c) 2019, The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+-->
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="com.android.theme.darkui.charcoal"
+    android:versionCode="1"
+    android:versionName="1.0">
+    <overlay android:targetPackage="android" android:category="android.theme.customization.darkui" android:priority="1"/>
+    <application android:label="CharcoalBlack" android:hasCode="false"/>
+</manifest>
diff --git a/packages/overlays/CharcoalDarkUIOverlay/res/values-night/colors.xml b/packages/overlays/CharcoalDarkUIOverlay/res/values-night/colors.xml
new file mode 100644
index 00000000000..7360755c630
--- /dev/null
+++ b/packages/overlays/CharcoalDarkUIOverlay/res/values-night/colors.xml
@@ -0,0 +1,42 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+/**
+ * Copyright (c) 2020, CAOS Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+-->
+<resources>
+    <color name="notification_primary_text_color_dark">#f0e5e5e5</color>
+    <color name="notification_secondary_text_color_dark">#b3e5e5e5</color>
+    <color name="notification_default_color_dark">#f0e5e5e5</color>
+    <color name="notification_material_background_color">#f00f1012</color> <!-- material_grey_900 -->
+    <color name="primary_material_dark">#f00f1012</color> <!-- material_grey_900 -->
+    <color name="primary_material_settings">#f00f1012</color> <!-- material_grey_900 -->
+    <color name="primary_dark_material_dark">@android:color/black</color>
+    <color name="primary_dark_material_settings">#f00f1012</color> <!-- material_grey_850 -->
+    <color name="primary_dark_device_default_dark">#f00f1012</color> <!-- material_grey_850 -->
+    <color name="primary_dark_device_default_settings">#f00f1012</color> <!-- material_grey_850 -->
+    <color name="secondary_material_settings">#f01a1b1e</color> <!-- material_grey_800 -->
+    <color name="tertiary_material_settings">#f0202126</color>
+    <color name="quaternary_material_settings">#f0dadce0</color> <!-- material_grey_200 -->
+		<color name="notification_material_background_color">#f00f1012</color>
+    <color name="material_grey_900">#f00f1012</color>
+    <color name="material_grey_850">#f00f1012</color>
+    <color name="material_grey_800">#f01a1b1e</color>
+    <color name="material_grey_600">#f0202126</color>
+    <color name="material_grey_300">#f03c4043</color>
+    <color name="material_grey_200">#f0dadce0</color>
+    <color name="material_grey_100">#f0eeeeee</color>
+    <color name="material_grey_50">#f0e8eaed</color>
+</resources>
diff --git a/packages/overlays/MidnightDarkUIOverlay/Android.mk b/packages/overlays/MidnightDarkUIOverlay/Android.mk
new file mode 100644
index 00000000000..7bc4eee2d6f
--- /dev/null
+++ b/packages/overlays/MidnightDarkUIOverlay/Android.mk
@@ -0,0 +1,30 @@
+#
+#  Copyright 2019, The Android Open Source Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+LOCAL_PATH:= $(call my-dir)
+include $(CLEAR_VARS)
+
+LOCAL_RRO_THEME := MidnightDarkUI
+
+LOCAL_PRODUCT_MODULE := true
+
+LOCAL_SRC_FILES := $(call all-subdir-java-files)
+
+LOCAL_RESOURCE_DIR := $(LOCAL_PATH)/res
+
+LOCAL_PACKAGE_NAME := MidnightDarkUIOverlay
+LOCAL_SDK_VERSION := current
+
+include $(BUILD_RRO_PACKAGE)
diff --git a/packages/overlays/MidnightDarkUIOverlay/AndroidManifest.xml b/packages/overlays/MidnightDarkUIOverlay/AndroidManifest.xml
new file mode 100644
index 00000000000..b7528ef1fb2
--- /dev/null
+++ b/packages/overlays/MidnightDarkUIOverlay/AndroidManifest.xml
@@ -0,0 +1,24 @@
+<!--
+/**
+ * Copyright (c) 2019, The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+-->
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="com.android.theme.darkui.midnight"
+    android:versionCode="1"
+    android:versionName="1.0">
+    <overlay android:targetPackage="android" android:category="android.theme.customization.darkui" android:priority="1"/>
+    <application android:label="Midnight" android:hasCode="false"/>
+</manifest>
diff --git a/packages/overlays/MidnightDarkUIOverlay/res/values-night/colors.xml b/packages/overlays/MidnightDarkUIOverlay/res/values-night/colors.xml
new file mode 100644
index 00000000000..1f2c7879e57
--- /dev/null
+++ b/packages/overlays/MidnightDarkUIOverlay/res/values-night/colors.xml
@@ -0,0 +1,42 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+/**
+ * Copyright (c) 2020, CAOS Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+-->
+<resources>
+    <color name="notification_primary_text_color_dark">#f0e7faff</color>
+    <color name="notification_secondary_text_color_dark">#b3e7faff</color>
+    <color name="notification_default_color_dark">#f0e7faff</color>
+    <color name="notification_material_background_color">#f00d2029</color> <!-- material_grey_900 -->
+    <color name="primary_material_dark">#f00d2029</color> <!-- material_grey_900 -->
+    <color name="primary_material_settings">#f00d2029</color> <!-- material_grey_900 -->
+    <color name="primary_dark_material_dark">#f009181e</color>
+    <color name="primary_dark_material_settings">#f00d2029</color> <!-- material_grey_850 -->
+    <color name="primary_dark_device_default_dark">#f00d2029</color> <!-- material_grey_850 -->
+    <color name="primary_dark_device_default_settings">#f00d2029</color> <!-- material_grey_850 -->
+    <color name="secondary_material_settings">#f01b2e38</color> <!-- material_grey_800 -->
+    <color name="tertiary_material_settings">#f01e343f</color>
+    <color name="quaternary_material_settings">#f03c4043</color> <!-- material_grey_200 -->
+    <color name="notification_material_background_color">#f00d2029</color>
+    <color name="material_grey_900">#f00d2029</color>
+    <color name="material_grey_850">#f00d2029</color>
+    <color name="material_grey_800">#f01b2e38</color>
+    <color name="material_grey_600">#f01e343f</color>
+    <color name="material_grey_300">#f09db3be</color>
+    <color name="material_grey_200">#f09aadb7</color>
+    <color name="material_grey_100">#f0cde0ea</color>
+    <color name="material_grey_50">#f0e7faff</color>
+</resources>
-- 
2.17.1

